src/app.ts

```diff
@@ -13,28 +14,55 @@ import Techblog from "./Techblog";
 import Uptime from "./Uptime";
 import VersionChecker from "./VersionChecker";
 
+import { CommandList } from "./CommandController";
 import { fileBackedObject } from "./FileBackedObject";
 import { PersonalSettings } from "./PersonalSettings";
 import { SharedSettings } from "./SharedSettings";
 
 // Load and initialise settings
 const sharedSettings = fileBackedObject<SharedSettings>("settings/shared_settings.json");
 const personalSettings = fileBackedObject<PersonalSettings>("settings/personal_settings.json");
+const commandList = fileBackedObject<CommandList>("settings/command_list.json");
 const bot = new Botty(personalSettings, sharedSettings);
 
 // Load extensions
 const joinMessaging = new JoinMessaging(bot.client, sharedSettings);
 const versionChecker = new VersionChecker(bot.client, sharedSettings, "data/version_data.json");
 const logger = new Logger(bot.client, sharedSettings);
-const uptime = new Uptime(bot.client, sharedSettings, personalSettings, "data/uptime_data.json");
 const keyFinder = new KeyFinder(bot.client, sharedSettings, "data/riot_keys.json");
-const forum = new ForumReader(bot.client, sharedSettings, personalSettings, "data/forum_data.json", keyFinder);
-const autoReact = new AutoReact(bot.client, sharedSettings, "data/thinking_data.json", "data/ignored_react_data.json");
-const officeHours = new OfficeHours(bot.client, sharedSettings, "data/office_hours_data.json");
-const riotAPILibraries = new RiotAPILibraries(bot.client, personalSettings, sharedSettings);
+// const forum = new ForumReader(bot.client, sharedSettings, personalSettings, "data/forum_data.json", keyFinder);
 const techblog = new Techblog(bot.client, sharedSettings, "data/techblog_data.json");
-const info = new Info(bot.client, sharedSettings, "data/info_data.json", versionChecker);
-const apiStatus = new ApiStatus(bot.client, sharedSettings);
+const controller = new CommandController(bot.client, sharedSettings);
+
+// register commands
+controller.registerCommand(commandList.controller.toggle, controller.onToggle);
+controller.registerCommand(commandList.controller.help, controller.onHelp);
+
+const notes = new Info(sharedSettings, "data/info_data.json", versionChecker);
+controller.registerCommand(commandList.info.note, notes.onNote);
+controller.registerCommand(commandList.info.all, notes.onAll);
+
+const officeHours = new OfficeHours(bot.client, sharedSettings, "data/office_hours_data.json");
+controller.registerCommand(commandList.officeHours.ask, officeHours.onAsk);
```

Comment:
Replace all of the command handlers with <class>.<handler>,bind(<class>)? Then all the functions can be class functions again, and they wont have to be arrow functions.

Reply:
eg in the case of officeHours.onAsk, it would become officeHours.onAsk.bind(officeHours)